### ISSUE TITLE
Acc bug when using sparse label and custom sparse_categorical_crossentropy like loss?

### ISSUE THREAD

**Issue by ethanyxfang** [keras-team/keras#10211]
When using sparse label without sparse_categorical_crossentropy loss, the acc is calculated by categorical_accuracy.
It's not expected when a custom loss like:
```
def my_loss(y_true, y_pred): 
    loss = K.min(K.sparse_categorical_crossentropy(y_true, y_pred), 1) 
    return loss
```
In keras/keras/engine/training.py it only compares the losses.sparse_categorical_crossentropy.
```
elif self.loss_functions[i] == losses.sparse_categorical_crossentropy:
    # case: categorical accuracy/crossentropy
    # with sparse targets
    if metric in ('accuracy', 'acc'):
        metric_fn = metrics_module.sparse_categorical_accuracy
    elif metric in ('crossentropy', 'ce'):
        metric_fn = metrics_module.sparse_categorical_crossentropy
```
I think it should judge the label with K.is_sparse or at lease give a warning?

**Comment by LCorleone** [comment]
what a coincidence!
I found the same issue. is there a way to solve it?
